### PR TITLE
[CI] Check for TODO_IN_THIS_COMMIT

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Checks for `TODO_IN_THIS_COMMIT` comments which should not be committed to the main branch. Also, checks for misspellings. 

### Summary generated by Reviewpad on 19 Oct 23 17:07 UTC
This pull request includes the following changes:

1. Added a new file `.github/workflows/reviewdog.yml` which configures the `reviewdog` job.
2. Added a job `check_todo_in_this` to check for `TODO_IN_THIS_` comments and prevent merging to the `main` branch.
3. Added a job `check_misspell` to check for misspellings using the `reviewdog/action-misspell` action.
4. Modified the `go.mod` file to add the dependency `github.com/cosmos/cosmos-proto v1.0.0-beta.2`.
5. Updated the dependencies `google.golang.org/genproto` and `github.com/cosmos/cosmos-proto` to remove the indirect tag.

Overall, this pull request adds new workflows for code review and introduces a new dependency in the `go.mod` file.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ require (
 	cosmossdk.io/api v0.3.1
```
Comment (okdas) [inline go.mod:6]: I am not sure why this was changed, but it seems like go changes the file consistently for me.

File: localnet/kubernetes/anvil.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+# TODO_IN_THIS_COMMIT - test if this fails the CI
```
[the remark this answers is not in the document]
Comment (okdas) [reply localnet/kubernetes/anvil.yaml]: ^ this is how it works - CI starts a conversation

File: .github/workflows/reviewdog.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  check_todo_in_this_commit:
+    name: Check TODO_IN_THIS_COMMIT
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: okdas/action-fail-on-found@v1
```
Comment (bryanchriswhite) [inline .github/workflows/reviewdog.yml:13]: @okdas where is the code for this?

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Comment by okdas** [comment]
Hey @Olshansk and @bryanchriswhite - the GitHub action itself resides in a separate repo - https://github.com/okdas/action-fail-on-found. Not much code to show - it is just a simple [one-liner](https://github.com/okdas/action-fail-on-found/blob/master/entrypoint.sh#L12) though that uses `the_silver_searcherer` to find mentions in the code and pipes the output into the [reviewdog](https://github.com/reviewdog/reviewdog). 
Let me know if you want to transfer the repo of this action into `pokt-network` organization, so we can all maintain it. Originally, I didn't think this was necessary.

File: .github/workflows/reviewdog.yml (hunk 3 above)
Comment (okdas) [reply .github/workflows/reviewdog.yml:13]: Hey @Olshansk and @bryanchriswhite - the GitHub action itself resides in a separate repo - https://github.com/okdas/action-fail-on-found. Not much code to show - it is just a simple [one-liner](https://github.com/okdas/action-fail-on-found/blob/master/entrypoint.sh#L12) though that uses `the_silver_searcherer` to find mentions in the code and pipes the output into the [reviewdog](https://github.com/reviewdog/reviewdog). 
Let me know if you want to transfer the repo of this action into `pokt-network` organization, so we can all maintain it. Originally, I didn't think this was necessary.

Comment (Olshansk) [reply .github/workflows/reviewdog.yml:13]: Yea, let's move it over just so it's all in one place.

File: .github/workflows/reviewdog.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+name: reviewdog
+
+on:
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  check_todo_in_this_commit:
+    name: Check TODO_IN_THIS_COMMIT
```
Comment (Olshansk) [inline .github/workflows/reviewdog.yml:9]:
```suggestion
    name: Check TODO_IN_THIS_
```

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Review by Olshansk — APPROVED** [review]

**Pull request merged**